### ISSUE TITLE
Need help about inputs of LGATr with padding

### ISSUE THREAD

**Issue by lixfrank** [heidelberg-hepml/lorentz-gatr#14]
I'm interesting about using LGATr in multi-particles system, but got problem about the input with different number of particles.
The number of particles in my dataset is not fixed, but 1 ,2 ,3, ... for each items. So I use Padding Mask by default transformer architecture of PyTorch to make a common length of tensors, but ignore the padded values during the attention computation.
Can you please give me some advice about using Padding Mask in LGATr?

**Comment by spinjo** [comment]
Sounds like this is the same setup that we have in our toptagging experiment. We deal with this by representing the data as `torch_geometric.data.Data` objects (see https://github.com/heidelberg-hepml/lorentz-gatr/blob/main/experiments/toptagging/dataset.py) and then using xformers attention with a block-diagonal mask (see https://github.com/heidelberg-hepml/lorentz-gatr/blob/main/experiments/toptagging/wrappers.py). 

As you say, you can also use padding together with an attention mask that ignores padded tokens in the attention computation. In this case you can also use the default pytorch attention  https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html. See https://github.com/hqucms/weaver-core/blob/main/weaver/nn/model/ParticleTransformer.py for how to do this (with a fancy attention mask) on the jet-tagging task.

**Comment by lixfrank** [comment]
> Sounds like this is the same setup that we have in our toptagging experiment. We deal with this by representing the data as `torch_geometric.data.Data` objects (see https://github.com/heidelberg-hepml/lorentz-gatr/blob/main/experiments/toptagging/dataset.py) and then using xformers attention with a block-diagonal mask (see https://github.com/heidelberg-hepml/lorentz-gatr/blob/main/experiments/toptagging/wrappers.py).
> 
> As you say, you can also use padding together with an attention mask that ignores padded tokens in the attention computation. In this case you can also use the default pytorch attention https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html. See https://github.com/hqucms/weaver-core/blob/main/weaver/nn/model/ParticleTransformer.py for how to do this (with a fancy attention mask) on the jet-tagging task.

Thank you! 
I tried to use the default pytorch attention mask, but found it useless, meaning the output still affected by the input should be masked. Perhaps the mask used is incorrect.
I'm going to try block-diagonal mask, thank you again.
